src/test/kotlin/org/rust/ide/inspections/lints/RsUnusedImportInspectionTest.kt

```diff
@@ -662,20 +662,170 @@ class RsUnusedImportInspectionTest : RsInspectionsTestBase(RsUnusedImportInspect
         }
     """)*/
 
-    /*@MockEdition(CargoWorkspace.Edition.EDITION_2018)
-    fun `test private import used in child module`() = checkByText("""
-       mod foo {
-            mod bar {
-                pub struct S;
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test private import unused in child module`() = checkByText("""
+        mod foo {
+            pub struct S;
+        }
+        <warning descr="Unused import: `foo::S`">use foo::S;</warning>
+
+        mod bar {}
+    """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test private import used in child module (direct reference)`() = checkByText("""
+        mod foo {
+            pub struct S;
+        }
+        use foo::S;
+
+        mod bar {
+            fn fun(_: super::S) {}
+        }
+    """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test private import used in child module (named import)`() = checkByText("""
+        mod foo {
+            pub struct S;
+        }
+        use foo::S;
+
+        mod bar {
+            use super::S;
+            fn fun(_: S) {}
+        }
+    """)
+
+    @MockEdition(CargoWorkspace.Edition.EDITION_2018)
+    fun `test private import used in child module (glob import)`() = checkByText("""
+        mod foo {
+            pub struct S;
+        }
+        use foo::S;
+
+        mod bar {
+            use super::S;
```

Comment:
This test is the same as `test private import used in child module (named import)`. Probably this should be `use super::*`.